node/chainSimulator/components/nodeFacade.go

```diff
@@ -102,6 +102,7 @@ func (node *testOnlyProcessingNode) createFacade(configs config.Configs, apiInte
 		nodePack.WithNodeStopChannel(node.CoreComponentsHolder.ChanStopNodeProcess()),
 		nodePack.WithImportMode(configs.ImportDbConfig.IsImportDBMode),
 		nodePack.WithESDTNFTStorageHandler(node.ProcessComponentsHolder.ESDTDataStorageHandlerForAPI()),
+		nodePack.WithNativeESDT(configs.GeneralConfig.SovereignConfig.GenesisConfig.NativeESDT),
```

Comment:
This is not ok.
This option is only needed for sovereign node, not for normal node, which doesn't need to know about nativeESDT.
As I've suggested before, this should be an extra option for sovereign node **only**. This should be passed from main file of sovereign chain as extra option. In case of chain simulator, it also needs to be passed as extra option from `NewSovereignChainSimulator` constructor.

Reply:
Added `NativeESDT` in sovereign node factory constructor, this way is only in sovereign components